apps/backend/app/services/model_training/feature_selection_service.py

```
@@ -0,0 +1,811 @@
+    ) -> Dict[str, float]:
+        """Statistical tests (chi2 or f-test)"""
+        # Fill NaN values
+        X_filled = X.fillna(0)
+
+        # Ensure all values are non-negative for chi2
+        if problem_type == "classification":
+            # Make values non-negative
+            X_positive = X_filled - X_filled.min() + 1e-10
+            try:
+                scores, _ = chi2(X_positive, y)
+            except:
```

Ruff (0.14.10)

Do not use bare `except`

(E722)